### PR TITLE
Fixed bug leading to invalid accessTokens when using Questionmarks in ap...

### PR DESCRIPTION
**Pull request by etticat**
When you use a '?' in your api-calls the accesstoken gets appended with another '?' https://github.com/Thuzi/facebook-node-sdk/blob/master/fb.js#L268

This leads to a api-call like this one: https://graph.facebook.com/me/friends?limit=10?access_token=vSAEFfionhaWFwai...

You see multible '?' in the request and Facebook can not find the accesstoken anymore because of this

### CODE REVIEW

File: fb.js [hunk 1]
```diff
@@ -247,7 +247,13 @@
             if(method === 'post') {
                 body = '';
                 if(params.access_token) {
-                    uri += '?access_token=' + encodeURIComponent(params.access_token);
+				    if((uri.indexOf("?") != -1)){
```
Comment (prabirshrestha) [inline fb.js]: need space before `{`.
use `!==`

File: fb.js [hunk 2]
```diff
@@ -265,7 +271,12 @@
                     body = body.substring(0, body.length - 1);
                 }
             } else {
-                uri += '?';
+                if((uri.indexOf("?") != -1)){
```
Comment (prabirshrestha) [inline fb.js]: space before `{`.
use `!==`

**Comment by prabirshrestha** [comment]
can u show a sample code on how you are making the request?

**Comment by etticat** [comment]
sure. I do it like this: 

var FB = require('fb');
FB.options({timeout: 1, accessToken: 'access_token'});

FB.api('/me?fields=id,name', function (res) {
    if(res && res.error) {
        if(res.error.code === 'ETIMEDOUT') {
            console.log('request timeout');
        }
        else {
            console.log('error', res.error);
        }
    }
    else {
        console.log(res);
    }
});

According to the Graph explorer this is a valid request: 
https://developers.facebook.com/tools/explorer/145634995501895/?method=GET&path=me%3Ffields%3Did%2Cname&version=

With my fix it works.

**Comment by etticat** [comment]
What is the status about this?

**Comment by prabirshrestha** [comment]
@etticat I currently no longer work at thuzi. @CRuppert can u look at this.

You can always pass the access token and any other query string values  as parameter instead of URL which I would recommend. Yo can see this in the readme.

**Comment by louisameline** [comment]
:+1: I think this should be merged, thank you.

**Comment by etticat** [comment]
Can somebody please merge this. I don't want to use my own version just for this little fix.

**Pull request merged**